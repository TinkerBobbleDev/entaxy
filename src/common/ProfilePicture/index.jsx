/* eslint-disable no-console */
import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

const styles = {
  profilePicture: {
    width: 30
  }
}

const mapStateToProps = ({ user }) => {
  return { user }
}

const ProfilePicture = ({ user }) => {
  if (user.isAuthenticated) {
    return (
      <img
        src={user.pictureUrl}
        alt={user.name}
        style={styles.profilePicture}
      />
    )
  }

  return null
}

ProfilePicture.propTypes = {
  user: PropTypes.object.isRequired
}

export default connect(mapStateToProps)(ProfilePicture)
