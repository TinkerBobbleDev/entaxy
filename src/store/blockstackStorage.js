/* eslint-disable no-console */
import { getFile, putFile, isUserSignedIn } from 'blockstack'

const filename = 'entaxy.json'

export const loadState = () => {
  if (isUserSignedIn()) {
    return getFile(filename)
      .then(data => JSON.parse(data))
      .catch((error) => {
        console.log(`Error loading file ${filename}:`, error)
        return {}
      })
  }
  return Promise.resolve({})
}

export const saveState = (state) => {
  if (isUserSignedIn()) {
    return putFile(filename, JSON.stringify(state))
  }
  return undefined
}
