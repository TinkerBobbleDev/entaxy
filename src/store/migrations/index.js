import migration0 from './0'
import migration1 from './1'
import migration2 from './2'
import migration3 from './3'

export default {
  0: migration0,
  1: migration1,
  2: migration2,
  3: migration3
}
