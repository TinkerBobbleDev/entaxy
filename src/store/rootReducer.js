import { combineReducers } from 'redux'
import userReducer from './user/reducer'
import settingsReducer from './settings/reducer'
import transactionsReducer from './transactions/reducer'
import marketValuesReducer from './marketValues/reducer'

export default combineReducers({
  user: userReducer,
  settings: settingsReducer,
  transactions: transactionsReducer,
  marketValues: marketValuesReducer
})
