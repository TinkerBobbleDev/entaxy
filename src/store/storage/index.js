/* eslint-disable import/no-cycle */
import { snakeCase } from 'lodash'
import * as blockstackStorage from './blockstackStorage'
import * as localStorage from './localStorage'
import { showOverlay, hideOverlay } from '../settings/actions'
import { initialState as settingsInitialState } from '../settings/reducer'
import { initialState as accountsInitialState } from '../accounts/reducer'
import { initialState as transactionsInitialState } from '../transactions/reducer'
import { initialState as exchangeRatesInitialState } from '../exchangeRates/reducer'
import { initialState as budgetInitialState } from '../budget/reducer'

export const loadItem = (itemName, newState) => (dispatch, getState) => {
  let payload = newState
  if (itemName === 'settings') {
    const { settings } = getState()
    if (settings !== null && settings !== undefined) {
      payload = { ...newState, overlayMessage: settings.overlayMessage }
    }
  }
  dispatch({
    type: `LOAD_${snakeCase(itemName).toUpperCase()}`,
    payload
  })
}

export const loadAllItems = (newState) => (dispatch) => {
  [
    'settings',
    'accounts',
    'transactions',
    'exchangeRates',
    'budget'
  ].map((itemName) => dispatch(loadItem(itemName, newState[itemName] || {})))
}


export const loadState = () => async (dispatch, getState) => {
  if (getState().user === undefined) return {}
  let newState
  switch (getState().user.isAuthenticatedWith) {
    case 'guest':
      dispatch(showOverlay('Loading data from localStorage...'))
      newState = await localStorage.loadState()
      break
    case 'blockstack':
      dispatch(showOverlay('Loading data from Blockstack...'))
      newState = await blockstackStorage.loadState()
      break
    // no default
  }
  if (newState === undefined) newState = { settings: settingsInitialState }
  dispatch(loadAllItems({
    ...newState,
    settings: {
      ...newState.settings,
      snackbarMessage: null
    }
  }))
  return dispatch(hideOverlay())
}

export const resetState = () => (dispatch, getState) => {
  const { settings } = getState()

  // Preserve the existing currency and locale
  dispatch(loadAllItems({
    settings: {
      ...settingsInitialState,
      currency: settings.currency,
      locale: settings.locale
    },
    accounts: accountsInitialState,
    transactions: transactionsInitialState,
    exchangeRates: exchangeRatesInitialState,
    budget: budgetInitialState
  }))
}

export const saveState = () => (_, getState) => {
  let { user } = getState()
  if (user === undefined) user = {}
  switch (user.isAuthenticatedWith) {
    case 'guest':
      return localStorage.saveState(getState())
    case 'blockstack':
      return blockstackStorage.saveState(getState())
    default:
      return undefined
  }
}
