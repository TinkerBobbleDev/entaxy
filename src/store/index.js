import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunkMiddleware from 'redux-thunk'
import rootReducer from './rootReducer'

const middlewares = [thunkMiddleware]

if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger')
  middlewares.push(logger)
}

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middlewares))
)

if (module.hot) {
  module.hot.accept(rootReducer, () => {
    store.replaceReducer(rootReducer)
  })
}

export default store
