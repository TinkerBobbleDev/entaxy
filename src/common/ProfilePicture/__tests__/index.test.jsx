import React from 'react'
import renderer from 'react-test-renderer'
import store from '../../../store'
import ProfilePicture from '../'

describe('ProfilePicture', () => {
  it('matches snapshot with logged out user', () => {
    const component = renderer.create(<ProfilePicture store={store} />)
    expect(component.toJSON()).toBeNull()
  })

  it('matches snapshot with logged in user profile', () => {
    // Set the store currenct state
    const fetchUserData = jest.fn(() => {
      return {
        type: 'FETCH_USER_DATA',
        payload: {
          isAuthenticated: true,
          username: 'username',
          name: 'Tester',
          pictureUrl: 'some url'
        }
      }
    })
    store.dispatch(fetchUserData())
    const component = renderer.create(<ProfilePicture store={store} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
