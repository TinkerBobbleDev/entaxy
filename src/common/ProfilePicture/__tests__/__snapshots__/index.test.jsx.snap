// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePicture matches snapshot with logged in user profile 1`] = `
<img
  alt="Tester"
  src="some url"
  style={
    Object {
      "width": 30,
    }
  }
/>
`;
